alicloud/resource_alicloud_kvstore_instance.go

```diff
@@ -0,0 +1,358 @@
+			request.SecurityIps = strings.Join(expandStringList(d.Get("security_ips").(*schema.Set).List())[:], COMMA_SEPARATED)
+		} else {
+			return fmt.Errorf("Security ips cannot be empty")
+		}
+		// wait instance status is Normal before modifying
+		if err := client.WaitForRKVInstance(d.Id(), Normal, DefaultLongTimeout); err != nil {
+			return fmt.Errorf("WaitForInstance %s got error: %#v", Running, err)
+		}
+		if _, err := conn.ModifySecurityIps(request); err != nil {
+			return fmt.Errorf("Create security whitelist ips got an error: %#v", err)
+		}
+		// wait instance status is Normal after modifying
+		if err := client.WaitForRKVInstance(d.Id(), Normal, DefaultLongTimeout); err != nil {
+			return fmt.Errorf("WaitForInstance %s got error: %#v", Running, err)
+		}
+	}
+
+	d.Partial(false)
+	return resourceAlicloudKVStoreInstanceRead(d, meta)
+}
+
+func resourceAlicloudKVStoreInstanceRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AliyunClient)
+	instance, err := client.DescribeRKVInstanceById(d.Id())
+	if err != nil {
+		if NotFoundError(err) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error Describe rKV InstanceAttribute: %#v", err)
+	}
+	d.Set("instance_name", instance.InstanceName)
+	d.Set("instance_class", instance.InstanceClass)
+	d.Set("availability_zone", instance.ZoneId)
+	d.Set("instance_charge_type", instance.ChargeType)
+	d.Set("instance_type", instance.InstanceType)
+	d.Set("vswitch_id", instance.VSwitchId)
+	d.Set("engine_version", instance.EngineVersion)
+	d.Set("connection_domain", instance.ConnectionDomain)
+	d.Set("port", instance.Port)
+	d.Set("private_ip", instance.PrivateIp)
+	d.Set("security_ips", strings.Split(instance.SecurityIPList, COMMA_SEPARATED))
+
+	return nil
+}
+
+func resourceAlicloudKVStoreInstanceDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AliyunClient)
+
+	instance, err := client.DescribeRKVInstanceById(d.Id())
+	if err != nil {
+		if NotFoundError(err) {
+			return nil
+		}
+		return fmt.Errorf("Error Describe DB InstanceAttribute: %#v", err)
+	}
+	if PayType(instance.ChargeType) == Prepaid {
+		return fmt.Errorf("At present, 'Prepaid' instance cannot be deleted and must wait it to be expired and release it automatically")
+	}
+	request := r_kvstore.CreateDeleteInstanceRequest()
+	request.InstanceId = d.Id()
+
+	return resource.Retry(5*time.Minute, func() *resource.RetryError {
+		_, err := client.rkvconn.DeleteInstance(request)
+
+		if err != nil {
+			if IsExceptedError(err, InvalidKVStoreInstanceIdNotFound) {
+				return nil
+			}
+			return resource.RetryableError(fmt.Errorf("Delete DB instance timeout and got an error: %#v", err))
+		}
+
+		instance, err := client.DescribeRKVInstanceById(d.Id())
+		if err != nil {
+			if NotFoundError(err) {
+				return nil
+			}
+			return resource.NonRetryableError(fmt.Errorf("Error Describe DB InstanceAttribute: %#v", err))
+		}
+		if instance == nil {
+			return nil
+		}
+
```

Comment:
I think line 297-299 are useless and method DescribeRKVInstanceById can ensure it.

Reply:
It seems like without any update.